### PR TITLE
cli: add the postgres \dT alias for listing enums

### PR DESCRIPTION
**Pull request by rohany**
Teach the cli to translate \dT into a call to `SHOW ENUMS`.

Release note (cli change): Recognize the \dT alias for listing user
defined types.

Release note (sql change): Add the `SHOW TYPES` command to list all user
defined types.

### CODE REVIEW

**Comment by rohany** [comment]
cc @vy-ton no need to make an issue

File: pkg/cli/sql.go
Comment (knz) [inline pkg/cli/sql.go:1047]: This makes me a little sad: `\dT` should be about user-defined types; why restrict it to enums?

There should be a `SHOW TYPES` in SQL that does "the right thing", and used here. Maybe it's time to add it preemptively?

**Review by knz — COMMENTED** [review]
Generally ok but see comment below.

File: pkg/cli/sql.go
Comment (rohany) [reply pkg/cli/sql.go:1047]: Yeah i think that makes sense. I'll add it.

File: docs/generated/sql/bnf/stmt_block.bnf
Comment (knz) [inline docs/generated/sql/bnf/stmt_block.bnf:990]: needs a parse test

**Review by knz — APPROVED** [review]
LGTM modulo missing parse test

File: pkg/sql/delegate/show_types.go
Comment (knz) [inline pkg/sql/delegate/show_types.go:22]: IMHO I'd have done it the other way around: make SHOW TYPES add a column with the kind of the type (enum, domain etc)

then make SHOW ENUMS filter with `WHERE kind = 'enum'`

File: pkg/sql/parser/parse_test.go
Comment (rohany) [inline pkg/sql/parser/parse_test.go:513]: I believe I added a parse test here @knz

File: pkg/sql/delegate/show_types.go
Comment (knz) [reply pkg/sql/delegate/show_types.go:22]: isn't there an `information_schema` that would report on that anyway?

Comment (rohany) [reply pkg/sql/delegate/show_types.go:22]: I'm not sure -- the show enums query is powered by the pg_catalog tables. My thought on the matter is that this way is more straightforward. The ways that different kinds are collected are pretty different. It would be simpler to have separate "methods" for each of those and then just aggregate them here.

Comment (knz) [reply pkg/sql/delegate/show_types.go:22]: all right your call - I approved the thing anyway

**Pull request merged**
